src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.cs

```diff
@@ -0,0 +1,415 @@
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void ArrayCreationWithInitializer_PrimitiveType()
+        {
+            string source = @"
+class C
+{
+    public void F()
+    {
+        var a = /*<bind>*/new string[] { string.Empty }/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: System.String) (OperationKind.ArrayCreationExpression, Type: System.String[])
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+    IFieldReferenceExpression: System.String System.String.Empty (Static) (OperationKind.FieldReferenceExpression, Type: System.String)
+";
+            VerifyOperationTreeForTest<ArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void ArrayCreationWithInitializer_UserDefinedType()
+        {
+            string source = @"
+class M { }
+
+class C
+{
+    public void F()
+    {
+        var a = /*<bind>*/new M[] { new M() }/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M[])
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+    IObjectCreationExpression (Constructor: M..ctor()) (OperationKind.ObjectCreationExpression, Type: M)
+";
+            VerifyOperationTreeForTest<ArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void ArrayCreationWithInitializer_ImplicitlyTyped()
+        {
+            string source = @"
+class M { }
+
+class C
+{
+    public void F()
+    {
+        var a = /*<bind>*/new[] { new M() }/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M[])
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+    IObjectCreationExpression (Constructor: M..ctor()) (OperationKind.ObjectCreationExpression, Type: M)
+";
+            VerifyOperationTreeForTest<ImplicitArrayCreationExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")]
+        public void ArrayCreationWithInitializer_MultipleInitializersWithConversions()
+        {
+            string source = @"
+class C
+{
+    public void F()
+    {
+        var a = """";
+        var b = /*<bind>*/new[] { ""hello"", a, null }/*</bind>*/;
```

Comment:
Could you also include tests for arrays created with explicit sizes and content initializers, correct and incorrect?